internal/ir/xds.go

```diff
@@ -2088,10 +2088,10 @@ type RateLimit struct {
 // GlobalRateLimit holds the global rate limiting configuration.
 // +k8s:deepcopy-gen=true
 type GlobalRateLimit struct {
-	// TODO zhaohuabing: add default values for Global rate limiting.
-
 	// Rules for rate limiting.
 	Rules []*RateLimitRule `json:"rules,omitempty" yaml:"rules,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
+	// ClientCertificate holds the client certificate secret for envoy to use when establishing a TLS connection to the rate limit service.
+	ClientCertificate TLSCertificate `json:"clientCertificate" yaml:"clientCertificate"`
```

Comment:
can we make this top level field, instead of per route, will increase size

Reply:
@arkodg how about introducing a SharedSecrets structure in the IR? It can also be used for OIDC HMAC secret and the TLS secret for envoy-wasm HTTP server connections.

Reply:
sure, also dont mind if its directly added as top level fields to https://github.com/envoyproxy/gateway/blob/020d60af70ea46f989550a80f95d1049990c647f/internal/ir/xds.go#L135